src/Commands/core/WorkspacesCommands.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Drush\Commands\core;
+
+use Consolidation\OutputFormatters\StructuredData\RowsOfFields;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Utility\Token;
+use Drupal\workspaces\WorkspaceOperationFactory;
+use Drush\Attributes as CLI;
+use Drush\Commands\DrushCommands;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+final class WorkspacesCommands extends DrushCommands {
+
+  /**
+   * Constructs a WorkspacesCommands object.
+   */
+  public function __construct(
+    private readonly WorkspaceOperationFactory $factory,
+    private readonly EntityTypeManagerInterface $entityTypeManager,
+  ) {
+    parent::__construct();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('workspaces.operation_factory'),
+      $container->get('entity_type.manager'),
+    );
+  }
+
+  /**
+   * Publish a workspace.
+   */
+  #[CLI\Command(name: 'workspaces:publish')]
```

Comment:
We should use class constant instead of string for command name. See other drush core commands.